packages/artillery/lib/platform/aws-lambda/dependencies.js

```diff
@@ -197,15 +208,118 @@ const createAndUploadLambdaZip = async (
   await createZip(dirname, zipfile);
 
   artillery.log('Preparing AWS environment...');
-  const s3path = await _uploadLambdaZip(bucketName, zipfile);
-  debug({ s3path });
+  const s3Path = await _uploadLambdaZip(bucketName, zipfile);
+  debug({ s3Path });
 
   return {
-    s3path,
+    s3Path,
     bom
   };
 };
 
+async function _uploadFileToS3(item, testRunId, bucketName) {
+  const s3 = new AWS.S3();
+  const prefix = `tests/${testRunId}`;
+  let body;
+  try {
+    body = fs.readFileSync(item.orig);
+  } catch (fsErr) {
+    console.log(fsErr);
+  }
+
+  if (!body) {
+    return;
+  }
+
+  const key = prefix + '/' + item.noPrefix;
+
+  try {
+    await s3
+      .putObject({
+        Bucket: bucketName,
+        Key: key,
+        // TODO: stream, not readFileSync
+        Body: body
+      })
+      .promise();
+
+    debug(`Uploaded ${key}`);
+    return;
+  } catch (err) {
+    throw err;
+  }
+}
+
+async function _syncS3(bomManifest, testRunId, bucketName) {
```

Comment:
This (I think) would be the exact same logic as what we already have for Fargate. Would be better to use that code instead (refactoring it to make it reusable if needed) rather than duplicating some of it here?

Reply:
Yes that's the plan, but doing that right now risks potentially breaking Fargate and Lambda. To minimise the risk of the change, I'd rather refactor and unify that logic separately.